lib/yard/tags/default_factory.rb

```diff
@@ -73,6 +73,9 @@ def parse_tag_with_title_and_text(tag_name, text)
       end
 
       def parse_tag_with_types_name_and_default(tag_name, text)
+        if text.nil?
+          return nil
```

Comment:
1. I would expect to see a warning being generated either here or somewhere in this code path.
2. You can `return` instead of `return nil`
3. I would recommend adding a newline of space after the `end` of this block (code style aesthetic)